lightpath/main.py

```diff
@@ -25,10 +27,25 @@ def create_arg_parser():
                         help='Experimental endstation(s) to show in Lightpath')
     parser.add_argument('--debug', dest='debug', action='store_true',
                         help='Show the DEBUG logging stream')
+    parser.add_argument('--cfg', required=False, default=None,
+                        help='Configuration yaml file')
     return parser
 
 
-def main(db, hutches):
+def get_qapp():
+    """Returns the global QApplication, creating it if necessary."""
+    global qapp
+    if qapp is None:
+        if QApplication.instance() is None:
+            logger.debug("Creating QApplication ...")
+            qapp = QApplication([])
+        else:
+            logger.debug("Using existing QApplication")
+            qapp = QApplication.instance()
+    return qapp
+
+
+def main(db, hutches, cfg):
```

Comment:
Docstring needs updating. Should `db`, `hutches` become optional now that both can be loaded from the config file?

Why? Well, what if I want to do `import lightpath.main; lightpath.main.main(cfg='...')`

Reply:
On the other hand, cfg is optional if db and hutches are specified.  Requiring one of two argument (sets) in python is a bit weird, but I gave it a shot with `typing.overload`